### PR TITLE
Functions - Min Max Range

### PR DESCRIPTION
**Pull request by Abduliitr**
## Checklist
- Solution is tested on sphere engine <br/>
    `Note:` Please copy only the solution part. Do not copy/paste the template code. That way, any issues with template code can be caught early on.
    - [x] Author
    - [x] Reviewer
- PR title is correct - [PR title conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#title-for-a-pull-request)
    - [x] Author
    - [x] Reviewer
- Folder/File naming is correct - [Naming conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#naming-of-files-and-folders)
    - [x] Author
    - [x] Reviewer
- Coding conventions are followed - [Coding conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#coding-conventions)
    - [x] Author
    - [x] Reviewer
- Test cases (action item for **reviewer**) <br/>
    - To be ensured by checking the problem on sphere-engine
    - If any test cases are missing, please add them - [how to add new test cases](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#adding-new-test-cases)
    - [x] There are at least 2 test cases
    - [x] There is at least 1 test case that is different from the sample test cases

### CODE REVIEW

**Comment by Abduliitr** [comment]
Two issues with the given template:

1. In `ConvertToNumber` function:

Current code:
```
function ConvertToNumber(list){
    for(let each of list){
        Number(each)
    }
    return list
}
```

Should be:
```
function ConvertToNumber(list){
    for(let each in list){
        list[each]=Number(list[each])
    }
    return list
}
```

2. In printing the output:

Currently:
`console.log(minMaxRange)`

Should be:
`console.log(...minMaxRange)`

I have done these corrections in the PR. Please update the same in the widget.

**Comment by LVSsandeepVarma** [comment]
@Abduliitr  updated the template

**Comment by sailesh-ramayanam** [comment]
@satya-9 Some issues with problem description and template.
1. There is no explanation for the sample test case.
2. Parameter to the functions is `input_numbers`. Please convert it to `camelCase`.
3. Typo - if m1 <= m2 return list ~off~ of numbers (same in the next line as well).
4. In the documentation for `getNumbersInRange`, `If no such numbers exist return a list with -1 i.e [-1]` is not mentioned.

File: solutions/functions/min-max-range.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+
+// ## This function should return single integer. The integer should be maximum value of input list
+// ## Please fill the following function
+function maximumValue(input_numbers){
+    // write below this here
+    let maxValue = Number.NEGATIVE_INFINITY;
+    for(let i = 0; i < input_numbers.length; i++){
+        if(input_numbers[i] > maxValue) maxValue = input_numbers[i];
```
Comment (sailesh-ramayanam) [inline solutions/functions/min-max-range.js:18]: @Abduliitr Please use curly braces. Same for the other function.

File: solutions/functions/min-max-range.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+//  Please do not change anything below this line.
+function ConvertToNumber(list){
+    for(let each in list){
+        list[each]=Number(list[each])
+    }
+    return list
+}
+
+let list1 = readLine().split(" ")
```
Comment (sailesh-ramayanam) [inline solutions/functions/min-max-range.js:67]: Semicolons are missing. Template is updated. Please get the latest template.

**Review by sailesh-ramayanam — CHANGES_REQUESTED** [review]

File: solutions/functions/min-max-range.js (hunk 2 above)
Comment (Abduliitr) [reply solutions/functions/min-max-range.js:67]: Updated as per latest template.

File: solutions/functions/min-max-range.js (hunk 1 above)
Comment (Abduliitr) [reply solutions/functions/min-max-range.js:18]: Added.

**Comment by satya-9** [comment]
> @satya-9 Some issues with problem description and template.
> 
> 1. There is no explanation for the sample test case.
> 2. Parameter to the functions is `input_numbers`. Please convert it to `camelCase`.
> 3. Typo - if m1 <= m2 return list ~off~ of numbers (same in the next line as well).
> 4. In the documentation for `getNumbersInRange`, `If no such numbers exist return a list with -1 i.e [-1]` is not mentioned.

1. @sailesh-ramayanam  added explanation
2. Converted to camelCase
3. Typo has been adjusted
4. mentioned `If no such numbers exist return a list with -1 i.e [-1]' in this template

**Review by satya-9 — APPROVED** [review]

**Comment by sailesh-ramayanam** [comment]
@satya-9 In test case explanation `we have to return the list of numbers which are 3<=number<=23 from the third input list which are [6,5,1,3,8,9,2]` this line is ambiguous. Are you saying this is the 3rd list or this is the output? Please rephrase it.

**Comment by satya-9** [comment]
> @satya-9 In test case explanation `we have to return the list of numbers which are 3<=number<=23 from the third input list which are [6,5,1,3,8,9,2]` this line is ambiguous. Are you saying this is the 3rd list or this is the output? Please rephrase it.
@sailesh-ramayanam  Changed the expalantion as ` The minimum value from the first input list given is 3 and maximum value from the second input list is 23 . We have to return the list of numbers which satisfies the condition 3<=number<=23 from the third input list. So the resulting output is [6,5,3,8,9]  ` .Is this okay? please check this.

**Review by sailesh-ramayanam — APPROVED** [review]

**Pull request merged**